connector-microsoft-teams/src/main/java/com/redhat/cloud/notifications/connector/microsoft/teams/TeamsRouteBuilder.java

```diff
@@ -24,7 +24,6 @@ public class TeamsRouteBuilder extends EngineToConnectorRouteBuilder {
     public void configureRoute() {
         from(direct(ENGINE_TO_CONNECTOR))
                 .routeId(connectorConfig.getConnectorName())
-                .removeHeaders(CAMEL_HTTP_HEADERS_PATTERN)
```

Comment:
Don't we have to remove these headers to the same way we do it in the `EngineToConnectorRouterBuilder` class?

Reply:
The line removed here didn't do anything with the latest version of the code. It was needed a few months ago when `engine` sent notifications to the Teams pseudo-connector through REST.

Reply:
Before this PR, the headers management was a bit messy. Here's how headers work now:
- headers coming from Kafka are removed in `EngineToConnectorRouterBuilder` before the connectors start building/sending notifications to external services
- headers sent to external services are removed in `OutgoingCloudEventBuilder` before the connectors start sending the notifications history to the engine through Kafka
- each connectors is still free to use headers with no restrictions between these two boundaries

Reply:
![image](https://github.com/RedHatInsights/notifications-backend/assets/10584698/b7204883-0f3d-4722-b2ad-b543e508d2d0)

Reply:
I understand! Thank you very much for the detailed explanation!